src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -2,143 +2,183 @@
-        {
-            if (_inRoot)
+            // Set default configurations and platforms for sln file
+            foreach (var platform in new List<string>{ "Any CPU", "x64", "x86" })
             {
-                // The user requested all projects go to the root folder
-                return null;
+                solution.AddPlatform(platform);
             }
-
-            if (_relativeRootSolutionFolders != null)
+            foreach (var buildType in new List<string> { "Debug", "Release" })
             {
-                // The user has specified an explicit root
-                return _relativeRootSolutionFolders;
+                solution.AddBuildType(buildType);
             }
 
-            // We determine the root for each individual project
-            var relativeProjectPath = Path.GetRelativePath(
-                PathUtility.EnsureTrailingSlash(slnFile.BaseDirectory),
-                fullProjectPath);
+            SolutionFolderModel? solutionFolder = (!_inRoot && _solutionFolderPath != null)
+                ? solution.AddFolder(GetSolutionFolderPathWithForwardSlashes(_solutionFolderPath))
+                : null;
 
-            return GetSolutionFoldersFromProjectPath(relativeProjectPath);
+            foreach (var projectPath in projectPaths)
+            {
+                string relativePath = Path.GetRelativePath(Path.GetDirectoryName(solutionFileFullPath), projectPath);
+                try
+                {
+                    AddProjectWithDefaultGuid(solution, relativePath, solutionFolder);
+                }
+                catch (InvalidProjectFileException ex)
+                {
+                    Reporter.Error.WriteLine(string.Format(CommonLocalizableStrings.InvalidProjectWithExceptionMessage, projectPath, ex.Message));
+                }
+                catch (SolutionArgumentException ex) when (solution.FindProject(relativePath) != null || ex.Type == SolutionErrorType.DuplicateProjectName)
+                {
+                    Reporter.Output.WriteLine(CommonLocalizableStrings.SolutionAlreadyContainsProject, solutionFileFullPath, relativePath);
+                }
+            }
+            if (solution.SolutionProjects.Count > 1)
+            {
+                // https://stackoverflow.com/a/14714485
+                solution.RemoveProperties("HideSolutionNode");
+            }
+            await serializer.SaveAsync(solutionFileFullPath, solution, cancellationToken);
         }
 
-        private static bool IsPathInTreeRootedAtSolutionDirectory(string path)
+        private void AddProjectWithDefaultGuid(SolutionModel solution, string relativePath, SolutionFolderModel solutionFolder)
         {
-            return !path.StartsWith("..");
-        }
+            // Open project instance to see if it is a valid project
+            ProjectRootElement projectRootElement = ProjectRootElement.Open(relativePath);
```

Comment:
This is needed: 
- to verify that the project file is valid. This is outside the scope of vs-solutionpersistence, and needed to preserve existing behavior
- as a fallback for getting the project type guid from the file directly in cases where it is not recognized by vs-solutionpersistence